darwin/exporter/formats/yolo.py

```diff
@@ -0,0 +1,90 @@
+from functools import partial
+from pathlib import Path
+from typing import Any, Callable, Dict, Iterable, List, Tuple, Union
+
+from rich.console import Console
+from rich.theme import Theme
+
+from darwin.datatypes import (
+    Annotation,
+    AnnotationFile,
+    ConversionError,
+    VideoAnnotation,
+    YoloAnnotation,
+)
+
+
+def export(annotation_files: Iterable[AnnotationFile], output_dir: Path) -> None:
+    """
+    Exports the given ``AnnotationFile``s into the yolo format inside of the given ``output_dir``.
+
+    Parameters
+    ----------
+    annotation_files : Iterator[dt.AnnotationFile]
+        The ``AnnotationFile``s to be exported.
+    output_dir : Path
+        The folder where the new coco file will be.
+    """
+    console = Console(theme=_console_theme())
+
+    output_dir.mkdir(parents=True, exist_ok=True)
+    for annotation_file in annotation_files:
+        output_file_path = (output_dir / annotation_file.filename).with_suffix(".txt")
+        errors, yolo_annotations = _convert_file(annotation_file)
+
+        with open(output_file_path, "w") as f:
+            for yolo in yolo_annotations:
+                f.write(f"{yolo.annotation_class} {yolo.x} {yolo.y} {yolo.width} {yolo.height}")
+
+        for err in errors:
+            console.print(
+                f"Failed to convert: '{err.filename}'. Reason: {err.reason}. Annotation: {err.annotation}\n",
+                style="error",
+            )
+
+
+def _convert_file(file: AnnotationFile) -> Tuple[List[ConversionError], List[YoloAnnotation]]:
+    conversions: List[Union[ConversionError, YoloAnnotation]] = _to_yolo(file)
+    yolo_annotations: List[YoloAnnotation] = [x for x in conversions if isinstance(x, YoloAnnotation)]
+    errors: List[ConversionError] = [x for x in conversions if isinstance(x, ConversionError)]
+    return (errors, yolo_annotations)
+
+
+def _to_yolo(annotation_file: AnnotationFile) -> List[Union[ConversionError, YoloAnnotation]]:
+    create_yolo_annotation_with_file = partial(_create_yolo_annotation, annotation_file=annotation_file)
+    return _map_list(create_yolo_annotation_with_file, annotation_file.annotations)
```

Comment:
Looks a bit convoluted to me. Would something like this work instead?
```suggestion
    return [_create_yolo_annotation(annotation) for annotation in annotation_file.annotations]
```

Reply:
So, here is the thing, you are using list comprehensions to do a `map`. Basically, list comprehensions in any language are for the most part , nested `flat_maps`. Using a map is simply more visible because it represents less complexity.

Having this in mind, I personally do not favor this specific change. What I have is the equivalent of:

```
[map(create_yolo_annotation_with_file, annotation_file.annotations)]
```
Which I have abstracted away into a function that I can reuse everywhere (same strategy i used for superannotate and labelbox). 

If I were to use Python's full functional power, I would probably end up with something like:

```
pipe(
  map(create_yolo_annotation_with_file, annotation_file.annotations), 
  list
)
```

Which is debatable. This, if you catch my _wave_, will likely remind you of Elixir (only a little). 
I understand this is Python. Not Elixir. 

But I also defend that when using abstractions from any science, I ought to make them clearly visible/descriptive. 
I cannot think of anything more descriptive than `_map_list`.